src/model/mod.rs

```diff
@@ -31,6 +31,51 @@ pub trait AlleleFreqs: Debug {}
 impl AlleleFreqs for DiscreteAlleleFreqs {}
 impl AlleleFreqs for ContinuousAlleleFreqs {}
 
+/// An allele frequency range
+#[derive(Debug)]
+pub struct AFRange {
+    pub inner: Range<AlleleFreq>,
+    pub left_exclusive: bool,
+    pub right_exclusive: bool
+}
```

Comment:
Please remove the `pub` from inner and implement the Deref trait for AFRange. This way, we can directly iterate over AFRange and also get access to start and end.
Simply add
```
impl Deref for AFRange {
    type Target = Range<AlleleFreq>;

    fn deref(&self) -> &Range<AlleleFreq> {
        &self.inner
    }
}
```